src/Tizen.TV.UIControls.Forms/Renderer/GridViewRenderer.cs

```diff
@@ -0,0 +1,299 @@
+            (Element as IGridViewController)?.SendItemFocused(new GridViewFocusedEventArgs(context.Data, context.RealizedView, false));
+        }
+
+        void OnItemFocused(object sender, GenGridItemEventArgs e)
+        {
+            if (e.Item == null)
+                return;
+
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            if (context.RealizedView == null)
+                context.RealizedView = CreateOrGetRealView(context);
+            (Element as IGridViewController)?.SendItemFocused(new GridViewFocusedEventArgs(context.Data, context.RealizedView, true));
+        }
+
+        void OnItemRealized(object sender, GenGridItemEventArgs e)
+        {
+            e.Item.EmitSignal("elm,state,5p,scale", "");
+            e.Item.SetPartColor("effect", new ElmSharp.Color(0, 0, 0, 0));
+        }
+
+        GenGridItemEventArgs OnSmartEvent(IntPtr data, IntPtr obj, IntPtr info)
+        {
+            return new GenGridItemEventArgs
+            {
+                Item = _itemHandleDic.ContainsKey(info) ? _itemHandleDic[info] : null
+            };
+        }
+
+        void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            UpdateItemsSource();
+        }
+
+        void UpdateItemsSource()
+        {
+            Control.Clear();
+            _gengridItemDic.Clear();
+            _itemHandleDic.Clear();
+            _itemViewDic.Clear();
+
+            if (_collectionChanged != null)
+            {
+                _collectionChanged.CollectionChanged -= OnCollectionChanged;
+                _collectionChanged = null;
+            }
+
+            if (Element.ItemsSource == null)
+            {
+                return;
+            }
+
+            foreach (var item in Element.ItemsSource)
+            {
+                var context = new GengridItemContext
+                {
+                    Data = item,
+                    RealizedView = null,
+                };
+                var gridItem = Control.Append(_gridItemClass, context);
+                IntPtr handle = gridItem;
+                _itemHandleDic.Add(handle, gridItem);
+                _gengridItemDic.Add(item, gridItem);
+            }
+
+            if (Element.ItemsSource is INotifyCollectionChanged collection)
+            {
+                _collectionChanged = collection;
+                collection.CollectionChanged += OnCollectionChanged;
+            }
+        }
+
+        View CreateOrGetRealView(GengridItemContext context)
+        {
+            if (_itemViewDic.ContainsKey(context.Data))
+            {
+                return _itemViewDic[context.Data].RealizedView;
+            }
+            var realView = CreateView(Element.ItemTemplate);
+            realView.BindingContext = context.Data;
+            context.RealizedView = realView;
+            _itemViewDic.Add(context.Data, context);
+            return realView;
```

Comment:
Why do you use `_itemViewDic`?
```suggestion
            if (context.RealizedView != null)
            {
                return context.RealizedView;
            }
            var realView = CreateView(Element.ItemTemplate);
            realView.BindingContext = context.Data;
            context.RealizedView = realView;
            return realView;
```

Reply:
Seems to need set a parent
``` c#
 realView.Parent = Element;
```

Reply:
Thank you for your recommendation.